src/reducers/vis-state-updaters.js

```diff
@@ -1007,8 +1042,8 @@ export const receiveMapConfigUpdater = (
 
   // reset config if keepExistingConfig is falsy
   let mergedState = !keepExistingConfig ? resetMapConfigUpdater(state) : state;
-  mergedState = mergeFilters(mergedState, filters);
   mergedState = mergeLayers(mergedState, layers);
+  mergedState = mergeFilters(mergedState, filters);
```

Comment:
in order for polygon filters to be loaded correctly we need load layers before filters

Reply:
see above comment